tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-card.html

```diff
@@ -0,0 +1,308 @@
+    tooltip-columns="[[_tooltipColumns]]"
+    smoothing-enabled="[[smoothingEnabled]]"
+    smoothing-weight="[[smoothingWeight]]"
+    tooltip-sorting-method="[[tooltipSortingMethod]]"
+    ignore-y-outliers="[[ignoreYOutliers]]"
+    request-manager="[[requestManager]]"
+    runs="[[runs]]"
+    tag="[[tag]]"
+    tag-metadata="[[tagMetadata]]"
+    active="[[active]]"
+    data-url="[[_dataUrl]]"
+    log-scale-active="[[_logScaleActive]]"
+    process-data="[[_processData]]"
+  >
+  <div id="buttons">
+    <paper-icon-button
+      selected$="[[_expanded]]"
+      icon="fullscreen"
+      on-tap="_toggleExpanded"
+    ></paper-icon-button>
+    <paper-icon-button
+      selected$="[[_logScaleActive]]"
+      icon="line-weight"
+      on-tap="_toggleLogScale"
+      title="Toggle y-axis log scale"
+    ></paper-icon-button>
+    <paper-icon-button
+      icon="settings-overscan"
+      on-tap="_resetDomain"
+      title="Fit domain to data"
+    ></paper-icon-button>
+    <span style="flex-grow: 1"></span>
+    <template is="dom-if" if="[[showDownloadLinks]]">
+      <div class="download-links">
+        <paper-dropdown-menu
+          no-label-float="true"
+          label="run to download"
+          selected-item-label="{{_runToDownload}}"
+        >
+          <paper-menu class="dropdown-content" slot="dropdown-content">
+            <template is="dom-repeat" items="[[runs]]">
+              <paper-item no-label-float=true>[[item]]</paper-item>
+            </template>
+          </paper-menu>
+        </paper-dropdown-menu>
+        <a
+          download="run_[[_runToDownload]]-tag-[[tag]].csv"
+          href="[[_csvUrl(tag, _runToDownload)]]"
+        >CSV</a> <a
+          download="run_[[_runToDownload]]-tag-[[tag]].json"
+          href="[[_jsonUrl(tag, _runToDownload)]]"
+        >JSON</a>
+        </div>
+      </div>
+    </template>
+  </div>
+  </tf-scalar-chart>
+  <style>
+    :host {
+      height: 235px;
+      width: 330px;
+      margin: 5px;
+      display: block;
+    }
+
+    :host[_expanded] {
+      height: 400px;
+      width: 100%;
+    }
+
+    tf-card-heading {
+      margin-bottom: 10px;
+    }
+
+    #buttons {
+      display: flex;
+      flex-direction: row;
+    }
+
+    paper-icon-button {
+      color: #2196F3;
+      border-radius: 100%;
+      width: 32px;
+      height: 32px;
+      padding: 4px;
+    }
+
+    paper-icon-button[selected] {
+      background: var(--tb-ui-light-accent);
+    }
+
+    .download-links {
+      display: flex;
+      height: 32px;
+    }
+
+    .download-links a {
+      font-size: 10px;
+      align-self: center;
+      margin: 2px;
+    }
+
+    .download-links paper-dropdown-menu {
+      width: 100px;
+      --paper-input-container-label: {
+        font-size: 10px;
+      }
+      --paper-input-container-input: {
+        font-size: 10px;
+      }
+    }
+  </style>
+</template>
+<script>
+  import {addParams} from '../tf-backend/urlPathHelpers.js';
```

Comment:
Note: Sometime in Q4 we're most likely going to be refactoring the codebase to move these script tags out into separate .js or .ts files, so we can use rules_typescript. Doesn't need to happen now.

Reply:
OK. @wchargin noted advantages to keeping scripts inline to make them easier to follow, but it would be nice to support the BUILD rules.